mifosng-android/src/main/java/com/mifos/mifosxdroid/online/ClientChooseFragment.java

```diff
@@ -0,0 +1,125 @@
+import android.widget.AdapterView;
+import android.widget.ListView;
+
+import com.mifos.mifosxdroid.R;
+import com.mifos.mifosxdroid.adapters.ClientChooseAdapter;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentTransaction;
+import com.mifos.mifosxdroid.core.BaseFragment;
+import com.mifos.objects.SearchedEntity;
+import com.mifos.objects.client.Client;
+import com.mifos.objects.client.Page;
+import com.mifos.utils.Constants;
+import com.mifos.utils.MifosApplication;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import butterknife.ButterKnife;
+import android.content.SharedPreferences;
+import butterknife.InjectView;
+import butterknife.OnClick;
+import retrofit.Callback;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+import android.content.Context;
+
+/**
+ * Created by Nasim Banu on 15-Jan-2016.
+ */
+public class ClientChooseFragment extends BaseFragment implements AdapterView.OnItemClickListener {
+    private static final String TAG = ClientSearchFragment.class.getSimpleName();
+    public static final String PREFS_NAME = "MY_PREFS";
+
+    @InjectView(R.id.lv_clients)
+    ListView results;
+
+    private List<Client> clients = new ArrayList<>();
+    List<Integer> clientIds = new ArrayList<Integer>();
+    private ClientChooseAdapter adapter;
+
+    public static ClientChooseFragment newInstance() {
+        ClientChooseFragment clientChooseFragment = new ClientChooseFragment();
+        return clientChooseFragment;
+    }
+
+    public ClientChooseFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_client_choose, null);
+        ButterKnife.inject(this, rootView);
+        adapter = new ClientChooseAdapter(getContext(), clients, R.layout.list_item_client);
+        results.setAdapter(adapter);
+        results.setOnItemClickListener(this);
+        loadClients();
+        return rootView;
+    }
+
+
+    public void loadClients() {
+        ((MifosApplication) getActivity().getApplication()).api.clientService.listAllClients(new Callback<Page<Client>>() {
+            @Override
+            public void success(Page<Client> page, Response response) {
+
+                clients = page.getPageItems();
+                adapter.setList(clients);
+                adapter.notifyDataSetChanged();
+                if (!clients.isEmpty()) {
+                    Iterator<Client> iterator = clients.iterator();
+
+                    clientIds.clear();
+                    while (iterator.hasNext()) {
+                        Client client = iterator.next();
+
+                        clientIds.add(client.getId());
+                    }
+                }
+            }
+
+            @Override
+            public void failure(RetrofitError retrofitError) {
+
```

Comment:
What happens in case of failure ? You could show a toast so that at least user could retry and not remain clueless.